src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -0,0 +1,116 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.FindPatientPage;
+import org.openmrs.reference.page.RegistrationEditSectionPage;
+import org.openmrs.reference.page.RegistrationSummaryPage;
+import org.openmrs.uitestframework.test.TestData;
+
+public class EditDemographicsAndContactInfoSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+	private RegistrationEditSectionPage registrationEditSectionPage;
+	private RegistrationSummaryPage registrationSummaryPage;
+
+	@Before(RunTest.HOOK.SELENIUM_EDIT_DEMOGRAPHICS)
+	public void visitPatientDashboard() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord()
+				.waitForPage();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_EDIT_DEMOGRAPHICS)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Edit Registration Information from Patient dashboard")
+	public void loadRegistrationSummaryPage() {
+		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage
+				.goToRegistrationSummary().waitForPage();
+	}
+
+	@Then("the system loads Registration Summary Page")
+	public void systemloadsRegistrationEditPage() {
+		assertTrue(registrationSummaryPage.containsText("DEMOGRAPHICS"));
+	}
+
+	@When("a user clicks on Edit link on Registration Summary Page")
+	public void loadDemographicsEditSection() {
+		registrationEditSectionPage = (RegistrationEditSectionPage) dashboardPage
+				.clickOnEditPatient().waitForPage();
+	}
+
+	@And("the system loads the edit demographics section")
+	public void systemloadsEditDemographicSection() {
+		assertTrue(textExists("Demographics"));
+	}
+
+	@And("a user enters the demographics of the patient")
+	public void captureNewDemographics() {
```

Comment:
Hello @kdaud  and @sherrif10 this method is editing the demographics of a patient. As you can see, the test is entering new demographics without first clearing the existing ones, and the test passes fine. However, my question is, do I need to go to the registrationEditSectionPage and add methods to clear demographics so that I can use them here during editing or I can just leave it the way it is.  What would be the best case scenario? I need your guidance.

Reply:
Thanks @Irenyak1  for looking into this subject! Whenever editing patient demographics, the existing values are overridden which is pretty fine so I would go with the existing implementation!